docs/release-process.md

```diff
@@ -0,0 +1,309 @@
+
+1. Create a new periodic E2E prow job for the "next" kubernetes minor version.
+   * All Kops prow jobs are defined [here](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/kops)
+2. Create a new presubmit E2E prow job for the new release branch.
+3. Create a new milestone in the GitHub repo.
+4. Update [prow's milestone_applier config](https://github.com/kubernetes/test-infra/blob/dc99617c881805981b85189da232d29747f87004/config/prow/plugins.yaml#L309-L313) to update master to use the new milestone and add an entry for the new branch that targets master's old milestone.
+5. Create the new release branch in git and push it to the GitHub repo.
+
+## Update versions
+
+See [1.19.0-alpha.1 PR](https://github.com/kubernetes/kops/pull/9494) for example
+
+* Use the hack/set-version script to update versions:  `hack/set-version 1.20.0 1.20.1`
+
+The syntax is `hack/set-version <new-release-version> <new-ci-version>`
+
+`new-release-version` is the version you are releasing.
+
+`new-ci-version` is the version you are releasing "plus one"; this is used to avoid CI jobs being out of semver order.
+
+Examples:
+
+| new-release-version  | new-ci-version
+| ---------------------| ---------------
+| 1.20.1               | 1.20.2
+| 1.21.0-alpha.1       | 1.21.0-alpha.2
+| 1.21.0-beta.1        | 1.21.0-beta.2
+
+
+* Update the golden tests: `hack/update-expected.sh`
+
+* Commit the changes (without pushing yet): `git add -p && git commit -m "Release 1.X.Y"`
+
+## Check builds OK
+
+```
+rm -rf .build/ .bazelbuild/
+make ci
+```
+
+## Send Pull Request to propose a release
+
+```
+git push $USER
+hub pull-request
+```
+
+Wait for the PR to merge
+
+## Tag the branch
+
+(TODO: Can we automate this?  Maybe we should have a tags.yaml file)
+
+```
+git checkout master
+git fetch origin
+git reset --hard origin/master
+```
+
+Make sure you are on the correct commit, and not a newer one!
+
+```
+VERSION=$(tools/get_version.sh | grep VERSION | awk '{print $2}')
+echo ${VERSION}
+```
+
+```
+git tag v${VERSION}
+git show v${VERSION}
+```
+
+Double check it is the correct commit!
+
+```
+git push git@github.com:kubernetes/kops v${VERSION}
+git fetch origin # sync back up
+```
+
+
+## Wait for CI job to complete
+
+The staging CI job should now see the tag, and build it (from the
+trusted prow cluster, using Google Cloud Build).
+
+The job is here: https://testgrid.k8s.io/sig-cluster-lifecycle-kops#kops-postsubmit-push-to-staging
+
+It (currently) takes about 10 minutes to run.
+
+In the meantime, you can compile the release notes...
+
+## Compile release notes
+
+e.g.
+
+```
+git checkout -b relnotes_${VERSION}
+
+FROM=1.18.0
+TO=1.18.1
+DOC=1.18
+git log v${FROM}..v${TO} --oneline | grep Merge.pull | grep -v Revert..Merge.pull | cut -f 5 -d ' ' | tac  > /tmp/prs
+echo -e "\n## ${FROM} to ${TO}\n"  >> docs/releases/${DOC}-NOTES.md
+relnotes  -config .shipbot.yaml  < /tmp/prs  >> docs/releases/${DOC}-NOTES.md
+```
+
```

Comment:
Remember to also update the doc index with the new release notes page (or change beta to stable)